components/giftbox.vue

```diff
@@ -76,27 +76,72 @@
       loading: true,
       features: [],
       selectedFeature: undefined,
-      seenFeatures: new Set(),
     }),
+    computed: {
+      seenFeatures() {
+        return this.$cookies.get('seenFeatures') || {};
+      },
+      cookieOptions() {
+        return {
+          maxAge: 60 * 60 * 24 * 30,
+        };
+      },
+    },
     created() {
       this.fetchFeatures().then((res) => {
         this.features = res;
+        // sync the seenFeatures with the incomming features from the api call
+        this.syncSeenFeatures();
+        // if there are new features
         if (this.features.length > 0) {
           this.selectedFeature = this.features[0];
-          this.seenFeatures.add(this.selectedFeature.uid);
+          this.addToSeenFeatures(this.selectedFeature.uid);
         }
       });
     },
     methods: {
       async fetchFeatures() {
         this.loading = true;
-        const response = await fetch(
-          process.env.newFeaturesUrl
-        );
+        const response = await fetch(process.env.newFeaturesUrl);
         const { data } = await response.json();
         this.loading = false;
         return data;
       },
+      addToSeenFeatures(uid) {
+        const seenFeatures = this.seenFeatures;
+        seenFeatures[uid] = true;
+        this.$cookies.set('seenFeatures', seenFeatures, this.cookieOptions);
+      },
+      isFeatureSeen(uid) {
+        const seenFeatures = this.seenFeatures;
+        return !!seenFeatures[uid];
+      },
+      syncSeenFeatures() {
+        const seenFeatures = this.seenFeatures;
+
+        // if there are no features
```

Comment:
Nit: This comment is a bit redundant, same as `// loop over seenFeatures` below. Comments describing _what_ the code does are usually not that valuable and sometimes even distracting. Comments that describe _why_ the code does something on the other hand are very useful later on.